sympy/series/limits.py

```diff
@@ -71,8 +70,16 @@ def heuristics(e, z, z0, dir):
         if r:
             rv = e.func(*r)
             if rv is S.NaN:
-                return
-
+                if e != ratsimp(e):
```

Comment:
`If r :` in line 70. I tried testing locally. They do pass . Basically, I am checking if the `heuristics` generates an output for the case. If `rv` is `Nan` then I apply the `ratsimp` mathod and then calculate the limit. The cases hopefully passes in the first iteration itself. But,still I will add another `try:` statement just to be sure for any other case.